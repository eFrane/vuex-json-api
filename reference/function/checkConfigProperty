<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Json:Api for Vuex</title>
    <meta name="generator" content="VuePress 1.8.0">
    
    <meta name="description" content="Automagic Json:Api integration for Vuex">
    
    <link rel="preload" href="/assets/css/0.styles.a40eab0d.css" as="style"><link rel="preload" href="/assets/js/app.e99eafd0.js" as="script"><link rel="preload" href="/assets/js/2.0dafffee.js" as="script"><link rel="preload" href="/assets/js/56.a489dc25.js" as="script"><link rel="prefetch" href="/assets/js/10.d336de47.js"><link rel="prefetch" href="/assets/js/11.e33487a3.js"><link rel="prefetch" href="/assets/js/12.207fd843.js"><link rel="prefetch" href="/assets/js/13.4b10f9ae.js"><link rel="prefetch" href="/assets/js/14.c4f9ecb0.js"><link rel="prefetch" href="/assets/js/15.feffaccf.js"><link rel="prefetch" href="/assets/js/16.e35df4ee.js"><link rel="prefetch" href="/assets/js/17.33cc5882.js"><link rel="prefetch" href="/assets/js/18.e4c5cceb.js"><link rel="prefetch" href="/assets/js/19.d21db1cb.js"><link rel="prefetch" href="/assets/js/20.ef87c427.js"><link rel="prefetch" href="/assets/js/21.47a4ed11.js"><link rel="prefetch" href="/assets/js/22.8f8c3435.js"><link rel="prefetch" href="/assets/js/23.7c076f99.js"><link rel="prefetch" href="/assets/js/24.693ccdb2.js"><link rel="prefetch" href="/assets/js/25.24fb0583.js"><link rel="prefetch" href="/assets/js/26.3534469b.js"><link rel="prefetch" href="/assets/js/27.b1c1c7e9.js"><link rel="prefetch" href="/assets/js/28.214e5e3a.js"><link rel="prefetch" href="/assets/js/29.4eae7e3e.js"><link rel="prefetch" href="/assets/js/3.437a0d76.js"><link rel="prefetch" href="/assets/js/30.c68541b4.js"><link rel="prefetch" href="/assets/js/31.c5b5d70a.js"><link rel="prefetch" href="/assets/js/32.101a2e61.js"><link rel="prefetch" href="/assets/js/33.fbffc187.js"><link rel="prefetch" href="/assets/js/34.a7072604.js"><link rel="prefetch" href="/assets/js/35.6b774ad6.js"><link rel="prefetch" href="/assets/js/36.329aa466.js"><link rel="prefetch" href="/assets/js/37.43083f8a.js"><link rel="prefetch" href="/assets/js/38.202477f7.js"><link rel="prefetch" href="/assets/js/39.71fc6a10.js"><link rel="prefetch" href="/assets/js/4.bfe88385.js"><link rel="prefetch" href="/assets/js/40.10e1bf31.js"><link rel="prefetch" href="/assets/js/41.b293e285.js"><link rel="prefetch" href="/assets/js/42.7685c85a.js"><link rel="prefetch" href="/assets/js/43.aaac79ff.js"><link rel="prefetch" href="/assets/js/44.fa5a7134.js"><link rel="prefetch" href="/assets/js/45.48e7868f.js"><link rel="prefetch" href="/assets/js/46.17e1d2d0.js"><link rel="prefetch" href="/assets/js/47.e082a658.js"><link rel="prefetch" href="/assets/js/48.14e53295.js"><link rel="prefetch" href="/assets/js/49.387e35aa.js"><link rel="prefetch" href="/assets/js/5.cb249b63.js"><link rel="prefetch" href="/assets/js/50.c621793e.js"><link rel="prefetch" href="/assets/js/51.61d326ce.js"><link rel="prefetch" href="/assets/js/52.bd6504bb.js"><link rel="prefetch" href="/assets/js/53.195ec585.js"><link rel="prefetch" href="/assets/js/54.cd3cd80c.js"><link rel="prefetch" href="/assets/js/55.7a4ec32a.js"><link rel="prefetch" href="/assets/js/57.c577613e.js"><link rel="prefetch" href="/assets/js/58.66a93e9c.js"><link rel="prefetch" href="/assets/js/59.becdceb6.js"><link rel="prefetch" href="/assets/js/6.4155846f.js"><link rel="prefetch" href="/assets/js/60.9977140e.js"><link rel="prefetch" href="/assets/js/61.074d1cf4.js"><link rel="prefetch" href="/assets/js/62.25ed9d76.js"><link rel="prefetch" href="/assets/js/63.b9031c84.js"><link rel="prefetch" href="/assets/js/64.74f38c3d.js"><link rel="prefetch" href="/assets/js/65.fe7e5276.js"><link rel="prefetch" href="/assets/js/7.6ed4aaa9.js"><link rel="prefetch" href="/assets/js/8.58c932af.js"><link rel="prefetch" href="/assets/js/9.9a6f2cf9.js">
    <link rel="stylesheet" href="/assets/css/0.styles.a40eab0d.css">
  </head>
  <body>
    <div id="app" data-server-rendered="true"><div class="theme-container no-sidebar"><header class="navbar"><div class="sidebar-button"><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512" class="icon"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z"></path></svg></div> <a href="/" class="home-link router-link-active"><!----> <span class="site-name">Json:Api for Vuex</span></a> <div class="links"><div class="search-box"><input aria-label="Search" autocomplete="off" spellcheck="false" value=""> <!----></div> <nav class="nav-links can-hide"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav></div></header> <div class="sidebar-mask"></div> <aside class="sidebar"><nav class="nav-links"><div class="nav-item"><a href="/" class="nav-link">
  Home
</a></div><div class="nav-item"><a href="/reference/" class="nav-link router-link-active">
  Reference
</a></div><div class="nav-item"><a href="/usage/" class="nav-link">
  Usage
</a></div> <a href="https://github.com/eFrane/vuex-json-api" target="_blank" rel="noopener noreferrer" class="repo-link">
    GitHub
    <span><svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" focusable="false" x="0px" y="0px" viewBox="0 0 100 100" width="15" height="15" class="icon outbound"><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path> <polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg> <span class="sr-only">(opens new window)</span></span></a></nav>  <!----> </aside> <main class="page"> <div class="theme-default-content content__default"><p><a name="checkConfigProperty"></a></p> <h2 id="checkconfigproperty-config-property-isrequiredprop"><a href="#checkconfigproperty-config-property-isrequiredprop" class="header-anchor">#</a> checkConfigProperty(config, property, isRequiredProp)</h2> <p><strong>Kind</strong>: global function</p> <table><thead><tr><th>Param</th> <th>Type</th></tr></thead> <tbody><tr><td>config</td> <td><code>object</code></td></tr> <tr><td>property</td> <td><code>String</code></td></tr> <tr><td>isRequiredProp</td> <td><code>Boolean</code></td></tr></tbody></table></div> <footer class="page-edit"><!----> <!----></footer> <div class="page-nav"><p class="inner"><span class="prev">
      ‚Üê
      <a href="/reference/" class="prev router-link-active">
        Reference (JS)
      </a></span> <!----></p></div> </main></div><div class="global-ui"></div></div>
    <script src="/assets/js/app.e99eafd0.js" defer></script><script src="/assets/js/2.0dafffee.js" defer></script><script src="/assets/js/56.a489dc25.js" defer></script>
  </body>
</html>
