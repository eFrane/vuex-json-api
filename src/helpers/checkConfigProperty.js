import { hasOwn } from '../shared/utils'

/**
 *
 * @param {object} config
 * @param {String} property
 * @param {Boolean} isRequiredProp
 */
export function checkConfigProperty (config, property, isRequiredProp = true) {
  if (config !== null && hasOwn(config, property)) {
    return true
  }

  if ((config === null ||
    hasOwn(config, property) === false) &&
    !isRequiredProp) {
    return false
  }

  throw new Error('Missing configuration property: ' + property)
}
