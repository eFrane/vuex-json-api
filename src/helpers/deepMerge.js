/**
 * Performs a deep merge of `source` into `target`.
 * Mutates `target` only but not its objects and arrays.
 *
 * @author inspired by [jhildenbiddle](https://stackoverflow.com/a/48218209).
 *
 * @param {Object} target
 * @param {Object} source
 *
 * @return {Object} merged Orbject
 */

export function deepMerge (target, source) {
  const isObject = (obj) => obj && typeof obj === 'object'

  if (!isObject(target) || !isObject(source)) {
    return source
  }

  Object.keys(source).forEach(key => {
    const targetValue = target[key]
    const sourceValue = source[key]

    if (Array.isArray(targetValue) && Array.isArray(sourceValue)) {
      target[key] = targetValue.concat(sourceValue)
    } else if (isObject(targetValue) && isObject(sourceValue)) {
      target[key] = deepMerge(Object.assign({}, targetValue), sourceValue)
    } else {
      target[key] = sourceValue
    }
  })

  return target
}
