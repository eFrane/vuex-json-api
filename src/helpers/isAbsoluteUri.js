/**
 *
 * @param {String} uri
 */
export function isAbsoluteUri (uri) {
  if (uri.indexOf('http://') === 0) {
    return true
  }

  if (uri.indexOf('https://') === 0) {
    return true
  }

  if (uri.match(/\/\/.+/)) {
    return true
  }

  return false
}
