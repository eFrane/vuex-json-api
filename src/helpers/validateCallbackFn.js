export function validateCallbackFn (fn) {
  if (fn.constructor !== Function) {
    return false
  }

  return true
}
