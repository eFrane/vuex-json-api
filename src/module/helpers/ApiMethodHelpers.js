import { hasOwn } from '../../shared/utils'

function checkPropertyOnObject (obj, property) {
  return typeof obj === 'object' && hasOwn(obj, property)
}

/**
 * Check whether the api supports collections
 *
 * @param {object} apiMethods supported api methods (list, get, update, delete)
 */
function isCollection (apiMethods) {
  return checkPropertyOnObject(apiMethods, 'list')
}

function allowsModification (apiMethods) {
  return checkPropertyOnObject(apiMethods, 'update')
}

function allowsReplacement (apiMethods) {
  return checkPropertyOnObject(apiMethods, 'replace')
}

function allowsCreation (apiMethods) {
  return checkPropertyOnObject(apiMethods, 'create')
}

function allowsDeletion (apiMethods) {
  return checkPropertyOnObject(apiMethods, 'delete')
}

function hasRelated (apiMethods) {
  return checkPropertyOnObject(apiMethods, 'related')
}

export {
  allowsCreation,
  allowsDeletion,
  allowsModification,
  allowsReplacement,
  hasRelated,
  isCollection
}
