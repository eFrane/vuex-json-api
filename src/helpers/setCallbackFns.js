import { validateCallbackFn } from './validateCallbackFn'

/**
 *
 * @param {Array} callbacks
 */
export function setCallbackFns (callbacks) {
  if (typeof callbacks === 'undefined' ||
    callbacks.constructor !== Array ||
    callbacks.reduce((carry, cb) => validateCallbackFn(cb) && carry, true) === false) {
    throw new Error('You must pass an array of valid callback functions to this method')
  }

  return callbacks
}
