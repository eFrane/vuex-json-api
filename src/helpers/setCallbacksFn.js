import {validateCallbackFn} from './validateCallbackFn'

/**
 *
 * @param {Array} callbacks
 */
export function setCallbacksFn (callbacks) {
  if (typeof callbacks === 'undefined' || callbacks.constructor !== Array ||
    callbacks.reduce((carry, cb) => validateCallbackFn(cb) && carry, true) === false) {
    throw new Error('You must pass an array of callbacks to this method')
  }
  return callbacks
}
