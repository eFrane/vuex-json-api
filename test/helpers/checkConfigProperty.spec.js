import { checkConfigProperty } from '@/helpers/checkConfigProperty'

const myObj = {
  testProp: 42
}

it('returns true if the property exists and is required', () => {
  expect(checkConfigProperty(myObj, 'testProp', true)).toBeTruthy()
  expect(checkConfigProperty(myObj, 'testProp')).toBeTruthy()
})

it('returns false if the property doesn\'t exist and isn\'t required', () => {
  expect(checkConfigProperty(myObj, 'bar', false)).toBeFalsy()
})

it('throws an error if the property doesn\'t exist but is required', () => {
  function testRequiredNonExisting () {
    checkConfigProperty(myObj, 'bar', true)
  }

  function testRequiredNonExistingDefaultRequired () {
    checkConfigProperty(myObj, 'bar')
  }

  expect(testRequiredNonExisting).toThrowErrorMatchingSnapshot()
  expect(testRequiredNonExistingDefaultRequired).toThrowErrorMatchingSnapshot()
})
