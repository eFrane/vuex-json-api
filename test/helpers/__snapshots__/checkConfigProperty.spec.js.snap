// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws an error if the property doesn't exist but is required 1`] = `"Missing configuration property: bar"`;

exports[`throws an error if the property doesn't exist but is required 2`] = `"Missing configuration property: bar"`;
